smac_planner/src/smac_planner.cpp

```diff
@@ -207,9 +216,20 @@ void SmacPlanner::activate()
     _node->get_logger(), "Activating plugin %s of type SmacPlanner",
     _name.c_str());
   _raw_plan_publisher->on_activate();
-  smoother_debug1_pub_->on_activate();
-  smoother_debug2_pub_->on_activate();
-  smoother_debug3_pub_->on_activate();
+  _smoother_debug1_pub->on_activate();
+  _smoother_debug2_pub->on_activate();
+  _smoother_debug3_pub->on_activate();
+
+  // Init the downsampler and publisher, if needed
+  // We have to initialize here and not on SmacPlanner::configure() because we need _costmap to be properly
+  // initialized after localizing
+  if (shouldDownsample()) {
+    _costmap_downsampler = std::make_unique<CostmapDownsampler>();
```

Comment:
All of this except the `on_activate` in this if statement should be in configuration, not activation state.

Reply:
I haven't found a workaround to make it work as you suggested. The way I have set up things, the original costmap needs to be initialized when we call the initalize method. I tried putting the initialization in the configure(), but at that point we haven't localized and it seems the costmap is not initialized yet.

Another option, is to initialize the costmap publisher in the createPlan() method (only once, of course). I will try that

Reply:
The initialize can wait until activate, but the make_unique should definitely be in configure. After you initialize `_costmap_downsampler`, you can use it as a bool rather than the `shouldDownsample()` (or replace the inside of it with that). Ex `if (_costmap_downsampler)` because it'll be null otherwise.